### PR TITLE
Bug 1960559: drop APIExtensions v1beta1

### PR DESCRIPTION
**Pull request by vrutkovs**
All manifests have moved to v1.

This also bumps openshift/api (and updates k8s deps to 0.21.1 to be uniform) so that all apiextensions/v1beta1 would no longer be used.

### CODE REVIEW

**Comment by vrutkovs** [comment]
/hold

I'd prefer [RBACv1beta1 PR](https://github.com/openshift/cluster-version-operator/pull/565) to land first

**Comment by vrutkovs** [comment]
/test unit

Flake

**Comment by vrutkovs** [comment]
/retest

Failed to setup initial cluster in upgrade test

File: hack/test-prerequisites.go [hunk 1]
```diff
@@ -36,12 +36,12 @@ func main() {
 			if err != nil {
 				log.Fatalf("Unable to read %s: %v", path, err)
 			}
-			var crd v1beta1.CustomResourceDefinition
+			var crd v1.CustomResourceDefinition
 			if err := yaml.Unmarshal(data, &crd); err != nil {
```
Comment (sttts) [inline hack/test-prerequisites.go:40]: You don't check that the API version matches? That's not going to make you happy when it doesn't.

Comment (sttts) [reply hack/test-prerequisites.go:40]: ok, test only.

Comment (vrutkovs) [reply hack/test-prerequisites.go:40]: I don't think we do that in tests. In actual flow non-v1 CRDs would be treated as "unstructered".

@wking @LalatenduMohanty any advice here? I suppose we should be tracking how many manifests are being applied by unstructured handler and attempt to minimize that number. Probably we should be exporting a new metric?

Comment (wking) [reply hack/test-prerequisites.go:40]: My long-term goal is to completely drop the unstructured handler, and fail folks in CI if they add a new manifest type that we don't support.  Logging in the unstructured handler that we can look for in gathered CI assets would help get us there.

Comment (LalatenduMohanty) [reply hack/test-prerequisites.go:40]: > if they add a new manifest type that we don't support. Logging in the unstructured handler that we can look for in gathered CI assets would help get us there.

+1

**Review by wking — APPROVED** [review]
/lgtm

[` pods should successfully create sandboxes by other` and `etcdHighNumberOfLeaderChanges`][1] are unrelated.

/override ci/prow/e2e-agnostic-upgrade

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/566/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade/1404700765347385344

**Comment by wking** [comment]
[`The HAProxy router should set Forwarded headers appropriately`][1] is unrelated.

/override ci/prow/e2e-agnostic

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/566/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1404829352356483072

**Comment by wking** [comment]
This PR is unrelated to the other recent changes, so save some time and money with a manual merge to cut off the retesting.

**Pull request merged**